### ISSUE TITLE
New failure in blackbox tests in stratisd_cert.py

### ISSUE THREAD

**Issue by mulkieran** [stratis-storage/project#104]
```
Using block device(s) for tests: ['/dev/vdb1', '/dev/vdb2', '/dev/vdb3']
----------
Stratisd dbus timeout: 300
Test devices: /dev/vdb1 /dev/vdb2 /dev/vdb3
Executing blackbox test 'python3 stratisd_cert.py' against test devices...
test_blockdev_list (__main__.StratisCertify) ... ok
test_filesystem_create (__main__.StratisCertify) ... ok
test_filesystem_create_same_name (__main__.StratisCertify) ... FAIL
test_filesystem_destroy (__main__.StratisCertify) ... ok
test_filesystem_list_empty (__main__.StratisCertify) ... ok
test_filesystem_list_not_empty (__main__.StratisCertify) ... ok
test_filesystem_rename (__main__.StratisCertify) ... FAIL
test_filesystem_rename_same_name (__main__.StratisCertify) ... ok
test_filesystem_snapshot (__main__.StratisCertify) ... FAIL
test_get_managed_objects (__main__.StratisCertify) ... ok
test_pool_add_cache (__main__.StratisCertify) ... ok
test_pool_add_data (__main__.StratisCertify) ... FAIL
test_pool_create (__main__.StratisCertify) ... ok
test_pool_create_same_name (__main__.StratisCertify) ... ok
test_pool_destroy (__main__.StratisCertify) ... FAIL
test_pool_list_empty (__main__.StratisCertify) ... ok
test_pool_list_not_empty (__main__.StratisCertify) ... ok
test_stratisd_version (__main__.StratisCertify) ... ok

======================================================================
FAIL: test_filesystem_create_same_name (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 277, in test_filesystem_create_same_name
    pool_path = make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert obj_path_exists and return_code == 0
AssertionError

======================================================================
FAIL: test_filesystem_rename (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 220, in test_filesystem_rename
    pool_path = make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert obj_path_exists and return_code == 0
AssertionError

======================================================================
FAIL: test_filesystem_snapshot (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 248, in test_filesystem_snapshot
    pool_path = make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert obj_path_exists and return_code == 0
AssertionError

======================================================================
FAIL: test_pool_add_data (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 163, in test_pool_add_data
    pool_path = make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert obj_path_exists and return_code == 0
AssertionError

======================================================================
FAIL: test_pool_destroy (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 196, in test_pool_destroy
    make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert obj_path_exists and return_code == 0
AssertionError

----------------------------------------------------------------------
Ran 18 tests in 48.560s

FAILED (failures=5)
```

**Comment by mulkieran** [comment]
We now know that the errors are quite repeatable, and that the return code in each is always 1.

**Comment by mulkieran** [comment]
Note that one thing that was in the original blackbox tests and that we lack is the little chunk of code to unmount the metadata volume. This is used in the ```destroy_all``` method that is run after every test. It's not clear that that is relevant to our current problem, though.

**Comment by mulkieran** [comment]
Bingo: it's udev db every time:

```
Test devices: /dev/vdb1 /dev/vdb2 /dev/vdb3
Executing blackbox test 'python3 stratisd_cert.py' against test devices...
test_blockdev_list (__main__.StratisCertify) ... ok
test_filesystem_create (__main__.StratisCertify) ... ok
test_filesystem_create_same_name (__main__.StratisCertify) ... FAIL
test_filesystem_destroy (__main__.StratisCertify) ... ok
test_filesystem_list_empty (__main__.StratisCertify) ... ok
test_filesystem_list_not_empty (__main__.StratisCertify) ... ok
test_filesystem_rename (__main__.StratisCertify) ... FAIL
test_filesystem_rename_same_name (__main__.StratisCertify) ... ok
test_filesystem_snapshot (__main__.StratisCertify) ... FAIL
test_get_managed_objects (__main__.StratisCertify) ... ok
test_pool_add_cache (__main__.StratisCertify) ... ok
test_pool_add_data (__main__.StratisCertify) ... FAIL
test_pool_create (__main__.StratisCertify) ... ok
test_pool_create_same_name (__main__.StratisCertify) ... FAIL
test_pool_destroy (__main__.StratisCertify) ... ok
test_pool_list_empty (__main__.StratisCertify) ... ok
test_pool_list_not_empty (__main__.StratisCertify) ... ok
test_stratisd_version (__main__.StratisCertify) ... ok

======================================================================
FAIL: test_filesystem_create_same_name (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 279, in test_filesystem_create_same_name
    pool_path = make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert return_code == 0, "return_code: %s, error_msg: %s" % (return_code, msg)
AssertionError: return_code: 1, error_msg: Device /dev/vdb1 has an existing signature Udev db says stratis, disk meta says no

======================================================================
FAIL: test_filesystem_rename (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 222, in test_filesystem_rename
    pool_path = make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert return_code == 0, "return_code: %s, error_msg: %s" % (return_code, msg)
AssertionError: return_code: 1, error_msg: Device /dev/vdb1 has an existing signature Udev db says stratis, disk meta says no

======================================================================
FAIL: test_filesystem_snapshot (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 250, in test_filesystem_snapshot
    pool_path = make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert return_code == 0, "return_code: %s, error_msg: %s" % (return_code, msg)
AssertionError: return_code: 1, error_msg: Device /dev/vdb1 has an existing signature Udev db says stratis, disk meta says no

======================================================================
FAIL: test_pool_add_data (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 165, in test_pool_add_data
    pool_path = make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert return_code == 0, "return_code: %s, error_msg: %s" % (return_code, msg)
AssertionError: return_code: 1, error_msg: Device /dev/vdb1 has an existing signature Udev db says stratis, disk meta says no

======================================================================
FAIL: test_pool_create_same_name (__main__.StratisCertify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 188, in test_pool_create_same_name
    make_test_pool(pool_name)
  File "stratis-cli-77.77.77/tests/blackbox/stratisd_cert.py", line 41, in make_test_pool
    assert return_code == 0, "return_code: %s, error_msg: %s" % (return_code, msg)
```

**Comment by mulkieran** [comment]
Resolved by #394.
